procrastinate/tasks.py

```diff
@@ -106,39 +131,30 @@ def add_namespace(self, namespace: str) -> None:
             for alias in self.aliases
         ]
 
-    def __call__(self, *args, **kwargs: types.JSONValue) -> Any:
+    def __call__(self, *args: P.args, **kwargs: P.kwargs) -> Any:
         return self.func(*args, **kwargs)
 
     @property
     def full_path(self) -> str:
         return utils.get_full_path(self.func)
 
-    async def defer_async(self, **task_kwargs: types.JSONValue) -> int:
+    async def defer_async(self, *_: Args.args, **task_kwargs: Args.kwargs) -> int:
```

Comment:
Why do defer accept args ? I think it was purposely not the case?

Reply:
Not that I wanted it, but it is not possible to use `ParamSpec` *kwargs* without *args*. 

They explain the why in the PEP:
> Furthermore, because the default kind of parameter in Python ((x: int)) may be addressed both positionally and through its name, two valid invocations of a (*args: P.args, **kwargs: P.kwargs) function may give different partitions of the same set of parameters. Therefore, we need to make sure that these special types are only brought into the world together, and are used together, so that our usage is valid for all possible partitions.

Source: https://peps.python.org/pep-0612/#id1

Reply:
ok